src/main/java/controllers/FeedController.java

```diff
@@ -30,56 +29,171 @@ public FeedController(InOut inOut, LoginManager loginManager) {
-        try {
-            String postSelection = this.inOut.getInput(this.postsString);
-            Integer postNumber = Integer.parseInt(postSelection);
-            Post selectedPost = this.postsActionMap.get(postNumber);
-            this.displayPostInfo(selectedPost);
-            String postAction =  this.inOut.getInput(this.postActionPrompt);
-            this.runPostAction(selectedPost, Integer.parseInt(postAction));
-            this.displayPostInfo(selectedPost);
-        } catch (IOException e) {
-            this.inOut.setOutput("An error occurred: " + e);
-        } catch (NumberFormatException nfe) {
-            this.inOut.setOutput("You entered an invalid action input.");
-        }
 
+        // Generate the posts and ask to select one to like/comment
+        String postsString = this.generateFeedActions(feed.getDisplayedPosts());
+        this.selectOnePost(postsString, feed.getDisplayedPosts().size());
     }
 
     private void runPostAction(Post selectedPost, Integer postAction) {
         switch (postAction) {
             case 0:
                 selectedPost.addLike(this.loginManager.getCurrUser());
                 break;
+            case 1:
+                selectedPost.addComment(this.getUserComment());
+                break;
             default:
                 this.inOut.setOutput("You entered an invalid action.");
         }
     }
 
+    private String generateFeedActions(ArrayList<Post> posts) {
+        this.postsActionMap = new HashMap<>();
+        StringBuilder postsString = new StringBuilder("Enter a post number for a detailed view of that post: ");
+
+        for (int i=0; i < posts.size(); i++) {
+            this.postsActionMap.put(i, posts.get(i));
+            postsString.append("\n");
+            postsString.append(i).append(" ");
+            // Change getAuthorId to getAuthor
+            postsString.append(posts.get(i).getAuthorId());
+            postsString.append("'s ").append(posts.get(i).getRecipe().getTitle());
+        }
+        return postsString.toString();
+    }
+
+    private void selectOnePost(String postsString, int numOfDisplayedPosts) {
+        int postNumber = this.getSelectedPostInput(postsString, numOfDisplayedPosts);
+        Post selectedPost = this.postsActionMap.get(postNumber);
+        this.displayPostInfo(selectedPost);
+
+        int postAction = this.getPostActionInput();
+
+        this.runPostAction(selectedPost, postAction);
+        this.displayPostInfo(selectedPost);
+    }
+
+    private int getSelectedPostInput(String postsString, int numOfDisplayedPosts) {
+        int postNumber = 0;
+        boolean success = false;
+        while (!success) {
+            try {
+                do {
+                    String postSelection = this.inOut.getInput(postsString);
+                    postNumber = Integer.parseInt(postSelection);
+                } while (postNumber < 0 || postNumber > numOfDisplayedPosts - 1);
```

Comment:
Would it work to have this as the outside while condition, instead of using the success boolean?